File: 1516-the-k-th-lexicographical-string-of-all-happy-strings-of-length-n/the-k-th-lexicographical-string-of-all-happy-strings-of-length-n.cpp
class Solution {
public:

    void recur(int n,string ans,vector<string>&str)
    {
        if(ans.size()==n)
        {
            str.push_back(ans);
            return ;
        }
        for(char i='a';i<='c';++i)
        {
            if(ans.size() && ans.back()==i)
                continue;
            recur(n,ans+i,str);
        }
    }

    string getHappyString(int n, int k) {
        string ans="";
        vector<string>str;
        recur(n,ans,str);
        if(str.size()<k)
            return "";
        sort(str.begin(),str.end());
        return str[k-1];
    }
};

File: 1516-the-k-th-lexicographical-string-of-all-happy-strings-of-length-n/the-k-th-lexicographical-string-of-all-happy-strings-of-length-n.java
class Solution {

    void recur(int n,String ans,List<String>list)
    {
        if(ans.length()==n)
        {
            list.add(ans);
            return ;
        }
        for(char ch='a';ch<='c';++ch)
        {
            int len=ans.length();
            if(len!=0 && ans.charAt(len-1)==ch)
                continue;
            ans+=ch;
            // System.out.println(ans);
            recur(n,ans,list);
            ans=ans.substring(0,len);
        }
    }

    public String getHappyString(int n, int k) {
        String ans="";
        List<String>list=new ArrayList<>();
        recur(n,ans,list);
        if(list.size()<k)
            return "";
        Collections.sort(list);
        return list.get(k-1);
    }
}